pyrene/src/components/Form/Form.tsx

```diff
@@ -0,0 +1,258 @@
+/* eslint-disable react/static-property-placement */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-floating-promises */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import React from 'react';
+import { ValidationError, ObjectSchema } from 'yup';
+import { MultiselectOption, Filters, Options } from '../Filter/types';
+
+type ValuesOf<T> = T[keyof T];
+type KeysOf<T> = keyof T;
+
+type FormValues = Filters;
+type ToucheValues = {
+  [K in KeysOf<FormValues>]: boolean
+};
+type Errors = Record<string, string>;
+
+export interface FormProps {
+  initialValues?: FormValues,
+  onChange: (value: FormValues, setter: (fieldName: KeysOf<FormState['values']>, value: Options) => void) => void,
+  onSubmit?: (formState: FormValues) => Promise<void>,
+  render: (args: RenderPropsArgs) => JSX.Element,
+  validateOnFirstTouch?: boolean,
+  validationSchema?: ObjectSchema<any> | null,
+}
+
+export interface FormState {
+  values: FormValues,
+  touched: ToucheValues,
+  isSubmitting: boolean,
+}
+
+type RenderPropsArgs = {
+  values?: FormValues,
+  errors: Errors,
+  touched: FormState['touched'],
+  isSubmitting: FormState['isSubmitting'],
+  submitDisabled: boolean,
+  initField: (fieldName: string) => InputComponentProps,
+};
+
+type InputComponentProps = {
+  name: KeysOf<FormState['values']>,
+  value: ValuesOf<FormState['values']>,
+  invalid: boolean,
+  invalidLabel: string,
+  onChange: (value: string, event: React.ChangeEvent<any>) => void,
+  onBlur: (event: React.FocusEvent<any>) => void,
+};
+
+const getTouchedState = (initialValues: FormValues) => (
```

Comment:
Since this is only for computing the initial state, I would rename it to `getInitialTouchedState`

Reply:
@maximelovino Ok. I ll do so. This task was originally only to do the migration. But I can also do the renaming.